packages/dtslint/src/index.ts

```diff
@@ -142,10 +157,77 @@ async function runTests(
   await assertNpmIgnoreExpected(dirPath);
   assertNoOtherFiles(dirPath);
 
+  let implementationPackage;
+  let warnings: string[] | undefined;
+  let errors: string[] | undefined;
+  let hasNpmVersionMismatch = false;
+
+  if (!skipNpmChecks && !expectOnly) {
+    const attw = await import("@arethetypeswrong/core");
+    const typesPackageVersion = `${packageJson.libraryMajorVersion}.${packageJson.libraryMinorVersion}`;
+    const pkg = await tryPromise(
+      attw.resolveImplementationPackageForTypesPackage(packageJson.name, `${typesPackageVersion}.9999`),
+    );
+    if (pkg) {
+      const { packageName, packageVersion, tarballUrl } = pkg;
+      if (packageJson.nonNpm === true) {
+        (errors ??= []).push(
+          `Package ${packageJson.name} is marked as non-npm, but ${packageName} exists on npm. ` +
+            `If these types are being added to DefinitelyTyped for the first time, please choose ` +
+            `a different name that does not conflict with an existing npm package.`,
+        );
+      } else if (!packageJson.nonNpm) {
+        if (!satisfies(packageVersion, typesPackageVersion)) {
+          hasNpmVersionMismatch = true;
+          const isError = !npmVersionExemptions.has(packageDirectoryNameWithVersion);
+          const container = isError ? (errors ??= []) : (warnings ??= []);
+          container.push(
+            (isError
+              ? ""
+              : `Ignoring npm version error because ${packageDirectoryNameWithVersion} was failing when the check was added. ` +
```

Comment:
the wording "${package} was failing when the check was added" is a little confusing. Which check? What's the scope of "was added" -- to CI? injected into this run?

Unfortunately I don't have any ideas on how to improve this yet.